File: apps/etl/src/run-all.ts
import * as fs from 'fs';
import * as path from 'path';
import * as dotenv from 'dotenv';

dotenv.config({
  path: path.resolve(__dirname, "../../../.env"),
});

import { saveGroupToDb } from './dgpEtl';
import { saveLattesToDb } from './lattesEtl';
import { PROCESSED_DATA_DIR } from './commom/config';

async function runAll() {
    console.log("=== RUNNING ETL FOR ALL GROUPS AND RESEARCHERS ===");
    
    const dgpDir = path.join(PROCESSED_DATA_DIR, 'dgp');
    const lattesDir = path.join(PROCESSED_DATA_DIR, 'lattes');

    if (!fs.existsSync(dgpDir)) {
        console.error(`DGP directory not found: ${dgpDir}`);
        return;
    }

    const groupFiles = fs.readdirSync(dgpDir).filter(f => f.endsWith('.json'));
    console.log(`Found ${groupFiles.length} groups in processed-data.`);

    for (const groupFile of groupFiles) {
        const groupFilePath = path.join(dgpDir, groupFile);
        console.log(`\n--------------------------------------------`);
        console.log(`[ETL-ALL] Processing Group: ${groupFile}`);
        
        const groupData = JSON.parse(fs.readFileSync(groupFilePath, 'utf-8'));
        
        // 1. Process group
        await saveGroupToDb(groupData);

        // 2. Process members
        if (groupData.membros && Array.isArray(groupData.membros)) {
            console.log(`[ETL-ALL] Group has ${groupData.membros.length} members. Checking files...`);
            for (const membro of groupData.membros) {
                if (!membro.lattes) continue;
                const lattesFileName = `${membro.lattes.trim()}.json`;
                const lattesFilePath = path.join(lattesDir, lattesFileName);

                if (fs.existsSync(lattesFilePath)) {
                    console.log(`[ETL-ALL] Processing Researcher Lattes: ${membro.nome} (${membro.lattes})`);
                    const lattesData = JSON.parse(fs.readFileSync(lattesFilePath, 'utf-8'));
                    await saveLattesToDb(lattesData);
                } else {
                    console.log(`[ETL-ALL] Lattes file NOT found for: ${membro.nome} (${membro.lattes})`);
                }
            }
        }
    }
    console.log("\n=== ETL RUN ALL COMPLETED ===");
}

runAll().catch(console.error);

File: etl/run_all.py
import os
import json
from db.client import db
from etl.common import PROCESSED_DATA_DIR
from etl.dgp import save_group_to_db
from etl.lattes import save_lattes_to_db

async def run_all():
    print("=== RUNNING ETL FOR ALL GROUPS AND RESEARCHERS ===")
    
    dgp_dir = os.path.join(PROCESSED_DATA_DIR, 'dgp')
    lattes_dir = os.path.join(PROCESSED_DATA_DIR, 'lattes')
    
    if not os.path.exists(dgp_dir):
        print(f"DGP directory not found: {dgp_dir}")
        return
        
    group_files = [f for f in os.listdir(dgp_dir) if f.endswith('.json')]
    print(f"Found {len(group_files)} groups in processed-data.")
    
    for group_file in group_files:
        group_file_path = os.path.join(dgp_dir, group_file)
        print(f"\n--------------------------------------------")
        print(f"[ETL-ALL] Processing Group: {group_file}")
        
        with open(group_file_path, "r", encoding="utf-8") as f:
            group_data = json.load(f)
            
        # 1. Process group
        await save_group_to_db(group_data)
        
        # 2. Process members
        membros = group_data.get("membros", [])
        if isinstance(membros, list):
            print(f"[ETL-ALL] Group has {len(membros)} members. Checking files...")
            for membro in membros:
                lattes_id = membro.get("lattes")
                if not lattes_id:
                    continue
                lattes_file_name = f"{lattes_id.strip()}.json"
                lattes_file_path = os.path.join(lattes_dir, lattes_file_name)
                
                if os.path.exists(lattes_file_path):
                    print(f"[ETL-ALL] Processing Researcher Lattes: {membro.get('nome')} ({lattes_id})")
                    with open(lattes_file_path, "r", encoding="utf-8") as f_lat:
                        lattes_data = json.load(f_lat)
                    await save_lattes_to_db(lattes_data)
                else:
                    print(f"[ETL-ALL] Lattes file NOT found for: {membro.get('nome')} ({lattes_id})")
                    
    print("\n=== ETL RUN ALL COMPLETED ===")
